aya/src/programs/lirc_mode2.rs

```diff
@@ -71,7 +72,11 @@ impl LircMode2 {
             }
         })?;
 
-        self.data.links.insert(LircLink::new(prog_fd, lircdev_fd))
+        // TODO (AM)
+        self.data.links.insert(LircLink::new(
+            prog_fd.try_clone_to_owned()?,
```

Comment:
I am not sure if cloning the fd is the best but the `query` function below has an `OwnedFd` from `query(target_fd, BPF_LIRC_MODE2, 0, &mut None)` so that was the easiest thing to do. We could change the API for `query` and use `BorrowedFd<'_>` in `LircLink` but that that may be more awkward due to lifetimes.